lib/core/src/lib/prediction/components/content-enrichment-menu/content-enrichment-menu.component.ts

```diff
@@ -0,0 +1,93 @@
+import { MatTooltipModule } from '@angular/material/tooltip';
+import { MatMenuModule, MatMenuTrigger } from '@angular/material/menu';
+import { MatIconModule } from '@angular/material/icon';
+import { MatButtonModule } from '@angular/material/button';
+import { ConfigurableFocusTrap, ConfigurableFocusTrapFactory } from '@angular/cdk/a11y';
+import { TranslateModule } from '@ngx-translate/core';
+import { Prediction, ReviewStatus } from '@alfresco/js-api';
+import { IconComponent } from '../../../icon';
+import { LocalizedDatePipe } from '../../../pipes';
+import { PredictionService } from '../../services';
+
+@Component({
+    standalone: true,
+    selector: 'adf-content-enrichment-menu',
+    templateUrl: './content-enrichment-menu.component.html',
+    styleUrls: ['./content-enrichment-menu.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    imports: [MatProgressSpinnerModule, MatTooltipModule, MatButtonModule, IconComponent, MatMenuModule, MatIconModule, TranslateModule, LocalizedDatePipe]
+})
+export class ContentEnrichmentMenuComponent implements OnInit {
+
+    @Input()
+    prediction: Prediction;
+
+    @ViewChild('menuContainer', { static: false })
+    menuContainer: ElementRef;
+
+    @ViewChild('menuTrigger', { static: false })
+    menuTrigger: MatMenuTrigger;
+
+    focusTrap: ConfigurableFocusTrap;
+    confidencePercentage: number;
+    previousValue: any;
+    predictionValue: any;
+    predictionDateTime: Date;
+
+    constructor(private predictionService: PredictionService, private focusTrapFactory: ConfigurableFocusTrapFactory) {}
+
+    ngOnInit() {
+        this.confidencePercentage = this.prediction?.confidenceLevel * 100 || 0;
+        this.previousValue = this.isDefined(this.prediction?.previousValue) ? this.prediction.previousValue : '';
+        this.predictionValue = this.isDefined(this.prediction?.predictionValue) ? this.prediction.predictionValue : '';
+        this.predictionDateTime = this.prediction?.predictionDateTime || null;
+    }
+
+    onMenuOpen() {
+        if (this.menuContainer && !this.focusTrap) {
+            this.focusTrap = this.focusTrapFactory.create(this.menuContainer.nativeElement);
+        }
+    }
+
+    onRevert() {
+        this.predictionService.reviewPrediction(this.prediction.id, ReviewStatus.REJECTED).subscribe(() => {
+            this.predictionService.predictionStatusUpdated$.next({key: this.prediction.property, previousValue: this.previousValue});
+            this.menuTrigger.closeMenu();
+        });
+    }
+
+    onConfirm() {
+        this.predictionService.reviewPrediction(this.prediction.id, ReviewStatus.CONFIRMED).subscribe(() => {
+            this.predictionService.predictionStatusUpdated$.next({key: this.prediction.property});
+            this.menuTrigger.closeMenu();
+        });
+    }
+
+    onClosed() {
+        this.focusTrap.destroy();
+        this.focusTrap = null;
+    }
+
+    private isDefined(value: string): boolean {
```

Comment:
I guess the value should be `any` here? Also maybe this can be moved to some sort of helper/utility function to be generally available?

Reply:
right, thanks

Reply:
not sure about where can we move it?

Reply:
Maybe to `ObjectUtils`?

Reply:
moved, ty